crates/corro-agent/src/api/peer/mod.rs

```diff
@@ -1304,9 +1310,12 @@ pub async fn parallel_sync(
                 }
 
                 if needs.is_empty() {
-                    if let Err(e) = tx.finish().instrument(info_span!("quic_finish")).await {
+                    if let Err(e) = tx.finish() {
                         warn!("could not finish stream while sending sync requests: {e}");
+                    } else if let Err(e) = tx.stopped().instrument(info_span!("quic_stopped")).await {
+                        warn!("could not wait for stream stopped while sending sync requests: {e}");
                     }
```

Comment:
As stated in the description, quinn split finish into finish and stopped, the latter of which is now the future that needs to be waited on for any buffered sends to be flushed.